contracts/AddressProviderNG.vy

```diff
@@ -0,0 +1,290 @@
+    """
+    @notice Update description for an existing _id
+    @param _id Identifier to set the new description for
+    @param _description New description to set
+    """
+    assert msg.sender == self.admin  # dev: admin-only function
+    assert self.get_id_info[_id].addr != empty(address)  # dev: id is empty
+
+    # Update description:
+    self.get_id_info[_id].description = _description
+
+    # Update metadata (version, update time):
+    self._update_entry_metadata(_id)
+
+
+@external
+def update_tags(_id: uint256, _tags: DynArray[String[64], 20]):
+    """
+    @notice Update tags for an existing _id
+    @param _id Identifier to set the new description for
+    @param _tags New tags to set to
+    """
+    assert msg.sender == self.admin  # dev: admin-only function
+    assert self.get_id_info[_id].addr != empty(address)  # dev: id is empty
+
+    # Update tags:
+    self.get_id_info[_id].tags = _tags
+
+    # Update metadata (version, update time):
+    self._update_entry_metadata(_id)
+
+
+
+@external
+def remove_id(_id: uint256) -> bool:
+    """
+    @notice Unset an existing identifier
+    @dev An identifier cannot ever be removed, it can only have the
+         address unset so that it returns empty(address)
+    @param _id Identifier to unset
+    @return bool success
+    """
+    assert msg.sender == self.admin  # dev: admin-only function
+    assert self.get_id_info[_id].addr != empty(address)  # dev: inactive id
+
+    self.get_id_info[_id].addr = empty(address)
+    self.get_id_info[_id].last_modified = block.timestamp
+    self.get_id_info[_id].description = ''
+    self.get_id_info[_id].tags = []
+    self.get_id_info[_id].version = 0
+
+    self.num_entries -= 1
+
+    self.check_id_exists[_id] = False
+
+    # Emit 0 in version to notify removal of id:
+    log EntryModified(_id, 0)
+
+    return True
+
+
+# ------------------------------ Admin Methods -------------------------------
+
+
+@external
+def commit_transfer_ownership(_new_admin: address) -> bool:
+    """
+    @notice Initiate a transfer of contract ownership
+    @dev Once initiated, the actual transfer may be performed three days later
+    @param _new_admin Address of the new owner account
+    @return bool success
+    """
+    assert msg.sender == self.admin  # dev: admin-only function
+    assert self.transfer_ownership_deadline == 0  # dev: transfer already active
+
+    deadline: uint256 = block.timestamp + 3*86400
+    self.transfer_ownership_deadline = deadline
+    self.future_admin = _new_admin
+
+    log CommitNewAdmin(deadline, _new_admin)
+
+    return True
+
+
+@external
+def apply_transfer_ownership() -> bool:
+    """
+    @notice Finalize a transfer of contract ownership
+    @dev May only be called by the current owner, three days after a
+         call to `commit_transfer_ownership`
+    @return bool success
+    """
+    assert msg.sender == self.admin  # dev: admin-only function
```

Comment:
Shouldn't this be called by the new_admin to accept ownership instead?
Forcing the new admin address to accept the transfer for the transfer to go through ensures that the operation doesn't accidentally burn the admin address by making admin a new address that no one controls.
The current setup doesn't seem to really bring any added security